data/blog/preact-or-svelte-an-embedded-widget-use-case.md

```diff
@@ -0,0 +1,116 @@
+
+## Svelte
+
+Svelte boasts an impressively small [bundle size of just 2.7kB](https://bundlephobia.com/package/svelte@4.2.9), the most compact among the frameworks under consideration. Replicating the feedback dialog and form in Svelte was relatively straightforward since most challenges with reducing the feedback down to a few components and transitioning to a framework were addressed during the process with Preact. The learning curve for Svelte was also small. The biggest challenge I came across when creating a project with Svelte was that the recommended command to create a project, <code>npm create svelte@latest myapp</code>, used SvelteKit, a full stack framework. This realization happened after replicating the feedback widget, when running the production build to assess build size. There were a lot of files which could have SvelteKit logic, making it difficult to directly compare build sizes with the Preact implementation.
+
+Luckily, the process of changing from using SvelteKit to pure Svelte with Vite was straightforward, and the resulting Svelte build size is 7.47kB.
+
+![Build size of Svelte implemetation](/images/preact-or-svelte-an-embedded-widget-use-case/svelte-build.png)
+
+## The Decision
+
+Both the Svelte implementation and Preact implementation have similar build sizes, with Svelte having the smaller build size.
+
+![Bundle size visualizer for Preact build](/images/preact-or-svelte-an-embedded-widget-use-case/preact-build-analyzer.png)
+
+_Vite bundle visualizer for the Preact build — almost half the build is just the framework_
+
+![Bundle size visualizer for Svelte build](/images/preact-or-svelte-an-embedded-widget-use-case/svelte-build-analyzer.png)
+
+_Vite bundle visualizer for the Svelte build — majority of the build is the feedback dialog and form_
+
+After looking at the builds more closely with the Vite bundle visualizer tool, I noticed that nearly half of the Preact implementation's build size was attributed to the framework itself. Considering our future plans to incorporate screenshots and annotation capabilities, which would significantly increase the build size, the marginal difference in framework size would become negligible, so the slightly smaller build size with Svelte would not help with the final bundle size. Additionally, Preact closely resembles React, which aligns well with our developers' familiarity with React and would greatly improve code maintainability. As a result, we chose Preact as the framework for expanding our User Feedback product to include screenshot and annotation functionalities.
+
+## The Aftermath
+
+In the end, adding Preact and screenshotting functionality to our user feedback widget increased the bundle size of @sentry/browser with the user feedback integration by 22.93%, from 30.81KB to 37.88KB. Although it’s a pretty large bundle size increase, using Preact allowed us to go from writing code like this:
```

Comment:
I actually am concerned of calling it out as an increase of bundle size, being so big without taking the chance to talk about treeshaking. And how that increase assumes all code from the package will be used.

If someone installing `@sentry/browser` doesn't use the feedback code, it'll be not include any of that code. And that's unfortunately not very well understood.

here's a blog post about reducing bundle size and talking about treeshaking you could link to, if you decide to add a note in that area: https://blog.sentry.io/sentry-bundle-size-how-we-reduced-replay-sdk-by-35/

```suggestion
In the end, adding Preact and screenshotting functionality to our user feedback widget increased the bundle size of `[@sentry/browser](https://www.npmjs.com/package/@sentry/browser)` with the user feedback integration by 22.93%, from 30.81KB to 37.88KB. Although it’s a pretty large bundle size increase, using Preact allowed us to go from writing code like this:
```